test/utils_test.js

```diff
@@ -617,6 +859,120 @@ describe("utils", () => {
         });
       });
 
+      it("should populate defaults for nested dependencies in arrays", () => {
+        const schema = {
+          type: "array",
+          items: {
+            properties: {
+              foo: {
+                type: "object",
+                properties: {
+                  name: {
+                    type: "string",
+                  },
+                },
+                dependencies: {
+                  name: {
+                    oneOf: [
+                      {
+                        properties: {
+                          name: {
+                            type: "string",
+                          },
+                          grade: {
+                            type: "string",
+                            default: "A",
+                          },
+                        },
+                      },
+                    ],
+                  },
+                },
+              },
+            },
+          },
+        };
+        expect(getDefaultFormState(schema, [{ foo: { name: "Name" } }])).eql([
+          {
+            foo: {
+              name: "Name",
+              grade: "A",
+            },
+          },
+        ]);
+      });
+
+      it("should populate defaults for nested dependencies in arrays when matching enum values in oneOf", () => {
```

Comment:
Why did you add this test? (was there a time when "should populate defaults for nested dependencies in arrays" passed but this test failed?)